TESTS/mbed_drivers/ticker/main.cpp

```diff
@@ -144,9 +206,138 @@ void test_case_2x_callbacks() {
     greentea_parse_kv(_key, _value, sizeof(_key), sizeof(_value));
 
     TEST_ASSERT_EQUAL_STRING_MESSAGE("pass", _key,"Host side script reported a fail...");
+    test_running = false;
+}
+
+/** Test many tickers run one after the other
+
+    Given a 16 Tickers
+    When schedule them one after the other with the same time intervals
+    Then tickers properly execute callbacks
+    When schedule them one after the other with the different time intervals
+    Then tickers properly execute callbacks
+ */
+void test_multi_ticker(void)
+{
+    Ticker ticker[TICKER_COUNT];
+
+    multi_counter = 0;
+    for (int i = 0; i < TICKER_COUNT; i++) {
+        ticker[i].attach_us(callback(wait_callback_multi), 100 * 1000);
+    }
+
+    Thread::wait(105);
+    for (int i = 0; i < TICKER_COUNT; i++) {
+            ticker[i].detach();
+    }
+
+    TEST_ASSERT_EQUAL(TICKER_COUNT, multi_counter);
+
+    multi_counter = 0;
+    for (int i = 0; i < TICKER_COUNT; i++) {
+        ticker[i].attach_us(callback(wait_callback_multi), (100 + i) * 1000);
+    }
+
+    Thread::wait(120);
+    for (int i = 0; i < TICKER_COUNT; i++) {
+        ticker[i].detach();
+    }
+    TEST_ASSERT_EQUAL(TICKER_COUNT, multi_counter);
+}
+
+/** Test multi callback time
+
+    Given a Ticker
+    When the callback is attached multiple times
+    Then ticker properly execute callback multiple times
+ */
+void test_multi_call_time(void)
+{
+    Ticker ticker;
+    int time_diff;
+
+    for (int i = 0; i < 10; i++) {
+        ticker_callback_flag = false;
+        gtimer.reset();
 
+        gtimer.start();
+        ticker.attach_us(callback(wait_callback_flag), 100000);
+        while(!ticker_callback_flag);
+        time_diff = gtimer.read_us();
+
+        TEST_ASSERT_UINT32_WITHIN(TOLERANCE, 100000, time_diff);
+    }
 }
 
+/** Test if detach cancel scheduled callback event
+
+    Given a Ticker with callback attached
+    When the callback is detached
+    Then the callback is not being called
+ */
+void test_detach(void)
+{
+    Ticker ticker;
+    int32_t ret;
+
+    ticker.attach(callback(wait_callback_sem), 0.1f);
```

Comment:
The semaphore can be local and there is no need to create a new function to release the semaphore: 

```c++
Semaphore sem(0, 1);

ticker.attach(callback(&sem, &Semaphore::release), 0.1f);
```

Reply:
but `attach` expects `Callback<void()> func` and `Semaphore::release` return `osStatus`

Reply:
changed to local semaphore passed as pointer
```
Semaphore sem(0, 1);
ticker.attach(callback(sem_release, &sem), 0.1f);
```